apps/ai-agent/app/guardrails.py

```
@@ -0,0 +1,14 @@
+from .schema import AgentDecision, DecideRequest
+
+ALLOWED = {'stop', 'delay', 'one_click', 'escalate_hitl'}
+def validate_decision(raw: dict, request: DecideRequest, taxonomy: dict) -> AgentDecision:
+    if set(raw) - set(AgentDecision.model_fields): raise ValueError('invented action or extra key')
+    result = AgentDecision.model_validate(raw)
+    if result.decision_type not in ALLOWED: raise ValueError('invalid decision')
+    if taxonomy.get('failure_type') == 'hard' and result.decision_type == 'one_click': raise ValueError('hard failure cannot one-click')
+    if taxonomy.get('failure_type') == 'hard': result.failure_type = 'hard'
```

Ruff (0.16.3)

[error] Multiple statements on one line (colon)

(E701)